src/ray/raylet/local_task_manager.cc

```diff
@@ -295,16 +295,40 @@ void LocalTaskManager::DispatchScheduledTasksToWorkers() {
         }
       }
 
+      // Check if the node is still schedulable. It may not be if dependency resolution
+      // took a long time.
+      auto allocated_instances = std::make_shared<TaskResourceInstances>();
+      bool schedulable =
+          !cluster_resource_scheduler_.GetLocalResourceManager().IsLocalNodeDraining() &&
+          cluster_resource_scheduler_.GetLocalResourceManager()
+              .AllocateLocalTaskResources(spec.GetRequiredResources().GetResourceMap(),
+                                          allocated_instances);
+      if (!schedulable) {
+        // The local node currently does not have the resources to run the task, so we
+        // should try spilling to another node.
+        bool did_spill = TrySpillback(work, is_infeasible);
```

Comment:
Inside of `TrySpillback`, we set the "preferred node ID" to be the current node. In the case where there is insufficient space to pin the local args, we probably _don't_ want to prefer the local node.

Should we check if there is sufficient space to pin the args at the top and thread this metadata through when considering spillback?